src/Core/Challenge/Dns/OvhSolver.php

```diff
@@ -0,0 +1,243 @@
+     */
+    private $endPoint;
+
+    /**
+     * @var string
+     */
+    private $consumerKey;
+
+    /**
+     * @var string
+     */
+    private $domainDelegation;
+
+    /**
+     * @param DnsDataExtractor $extractor
+     * @param ClientInterface  $client
+     */
+    public function __construct(
+        DnsDataExtractor $extractor = null,
+        ClientInterface $client = null
+        ) {
+        $this->extractor = null === $extractor ? new DnsDataExtractor() : $extractor;
+        $this->client = null === $client ? new Client() : $client;
+        $this->logger = new NullLogger();
+    }
+
+    /**
+     * Configure the service with a set of configuration.
+     */
+    public function configure(array $config)
+    {
+        $this->appKey = $config['app_key'];
+        $this->appSecret = $config['app_secret'];
+        $this->endPoint = $config['end_point'];
+        $this->consumerKey = $config['consumer_key'];
+        if (isset($config['domain_delegation'])) {
+            $this->domainDelegation = $config['domain_delegation'];
+        }
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function supports(AuthorizationChallenge $authorizationChallenge)
+    {
+        return 'dns-01' === $authorizationChallenge->getType();
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function solve(AuthorizationChallenge $authorizationChallenge)
+    {
+        return $this->solveAll([$authorizationChallenge]);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function solveAll(array $authorizationChallenges)
+    {
+        Assert::allIsInstanceOf($authorizationChallenges, AuthorizationChallenge::class);
+
+        foreach ($authorizationChallenges as $authorizationChallenge) {
+            $this->logger->debug('Create TXT for '.$authorizationChallenge->getDomain());
+
+            $topLevelDomain = $this->getTopLevelDomain($authorizationChallenge->getDomain());
+            $recordName = $this->extractor->getRecordName($authorizationChallenge);
+            $recordValue = $this->extractor->getRecordValue($authorizationChallenge);
+
+            $subDomain = \str_replace('.'.$topLevelDomain.'.', '', $recordName);
+
+            $client = new Client(['http_errors' => false]);
+
+            $ovh = new Api($this->appKey,
+                $this->appSecret,
+                $this->endPoint,
+                $this->consumerKey,
+                $client);
+
+            if ('' !== $this->domainDelegation) {
+                $topLevelDomain = $this->domainDelegation;
+            }
+
+            $result = $ovh->post('/domain/zone/'.$topLevelDomain.'/record', [
+            'fieldType' => 'TXT',
+            'subDomain' => $subDomain,
+            'target' => $recordValue,
+            'ttl' => 600,
+            ]);
+
+            if (isset($result['message']) && '' !== $result['message']) {
+                $this->logger->error('OVH Exception Post = '.$result['message'].' '.$topLevelDomain);
```

Comment:
You probably don't need to use the logger here as the monitoring tools will catch the exception